jme3-core/src/main/java/com/jme3/util/res/ResourcesLoader.java

```diff
@@ -0,0 +1,101 @@
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jme3.util.res;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+
+/**
+ * This class is used to load resources from the default location
+ * usually the classpath.
+ */
+public class ResourcesLoader {
+    public static final String PROPERTY_RESOURCES_LOADER_IMPLEMENTATION = "com.jme3.ResourcesLoaderImplementation";
+    private static final String DEFAULT_IMPL = "com.jme3.util.res.ResourcesLoaderJImpl";
+
+    private static final Logger LOGGER = Logger.getLogger(ResourcesLoader.class.getName());
+    private static ResourcesLoaderImpl impl = null;
+
+    private static ResourcesLoaderImpl getImpl() {
+        if (impl != null) return impl;
+        String className = System.getProperty(PROPERTY_RESOURCES_LOADER_IMPLEMENTATION, DEFAULT_IMPL);
+        try {
+            impl = (ResourcesLoaderImpl) Class.forName(className).getDeclaredConstructor().newInstance();
+        } catch (final Throwable e) {
+            LOGGER.log(Level.WARNING, "Unable to access {0}", className);
+            try {
+                impl = (ResourcesLoaderImpl) Class.forName(DEFAULT_IMPL).getDeclaredConstructor().newInstance();
+            } catch (final Throwable e1) {
+                throw new RuntimeException("Unable to access default resources loader implementation", e1);
+            }
+        }
+        return impl;
+    }
+    
+    public static void setImpl(ResourcesLoaderImpl impl) {
+        ResourcesLoader.impl = impl;
+    }
+
+    
+    public static URL getResource(String path) {
+        return getImpl().getResource(path, null);
+    }
+    
+    
+    public static URL getResource(String path, Class<?> parent) {
+        return getImpl().getResource(path, parent);
+    }
+    
+    public static InputStream getResourceAsStream(String path) {
+        return getImpl().getResourceAsStream(path, null);
+    }
+        
+    public static InputStream getResourceAsStream(String path,Class<?> clazz) {
+        return getImpl().getResourceAsStream(path,clazz);
+    }
+    
+
+   public static Enumeration<URL> getResources(String path ) throws IOException {
+        return getImpl().getResources(path,null);        
+    }
+
+
+   public static Enumeration<URL> getResources(String path,Class<?> clazz) throws IOException {
+        return getImpl().getResources(path,clazz);        
+    }
```

Comment:
Please fix the formatting issues in the above methods. (add required whitespace and remove extras)

Reply:
Done

Reply:
Thanks